### PR TITLE
Improve notebooks

### LINKED ISSUE #116: Make a testable notebook(s) for Zalando programmers
* At the end of the sprint, we need to provide current state to our testers, in order to receive feedback.
* Define which features exactly should be tested. 
* Explain how the results should be interpreted, with texts inside of the notebooks (it will be shown to security people).
* People can clone this repo, import our tool, and use it in code.
* Make sure that User Output class is integrated as well.

### PR DESCRIPTION
**Pull request by marisanest**
Closes #116.

### CODE REVIEW

**Comment by jtorhoff** [comment]
Hey I just changed the base branch from `main` to `team1sprint4`

File: notebooks/membership_inference_attack.ipynb [hunk 1]
```diff
@@ -76,16 +184,23 @@
...
    "outputs": [],
    "source": [
-    "attack.attack_output(x_train[:100], y_train[:100], np.ones((100,)))"
+    "output = attack.attack_output(\n",
+    "    x_train[:100], \n",
+    "    y_train[:100], \n",
+    "    np.ones((100,))\n",
+    ")\n",
+    "\n",
+    "output.to_json()"
```
Comment (Erdnaf) [inline notebooks/membership_inference_attack.ipynb:193]: Actualy when i run the notebook locally I get the error `AttributeError: 'UserOutputInferenceAttack' object has no attribute 'to_json'`. strangely it doses work on google colab and also if I just change it to `print(output)`.
Does someone has a explanation for that?
also throws that error, when trying to call `.to_dict()`

**Review by Erdnaf — APPROVED** [review]
Very good work, good structure and good coments 👍.
I had a small problem, but since it works on colab maybe the error is on my side

File: notebooks/membership_inference_attack.ipynb (hunk 1 above)
Comment (jtorhoff) [reply notebooks/membership_inference_attack.ipynb:193]: Maybe you have an outdated version of our package locally which the notebook uses somehow? Try uninstalling it and then run the notebook.

Comment (Erdnaf) [reply notebooks/membership_inference_attack.ipynb:193]: ah that actualy resolved it thanks. i just installed the new package over the old one, but uninstalling and reinstaling works

**Comment by jtorhoff** [comment]
Impressive work. The notebooks are thoroughly documented and run fine.

**Review by jtorhoff — APPROVED** [review]

**Review by budmil — APPROVED** [review]

**Comment by Friedrich-Mueller** [comment]
Lets please wait untill tomorrow for our daily sprint meeting
I think we have to optimize some things. But other than that I have to agree with the others. Its really nice work.

**Pull request merged**